graph/src/components/store.rs

```diff
@@ -1717,7 +1717,11 @@ impl LfuCache<EntityKey, Option<Entity>> {
     ) -> Result<Option<Entity>, QueryExecutionError> {
         match self.get(&key) {
             None => {
-                let entity = store.get(key.clone())?;
+                let mut entity = store.get(key.clone())?;
+                if let Some(entity) = &mut entity {
+                    // `__typename` is for queries not for mappings.
+                    entity.remove("__typename");
+                }
```

Comment:
As I said in slack, I think the right place to do this is in `HostExports::store_set` before we hand the entity to POI, and we should remove _everything_ that is not an attribute mentioned in the GraphQL schema - that way, we make sure that POI does not change if we ever need to add additional attributes into entities to support queries.

Reply:
I think the comment may not reflect your latest thinking, but I want to make clear for the record here as was discussed on slack that I strongly disagree with any change that moves code before the PoI event emit